fix: updated device data fetching using relational joins instead of v…

ThashmikaRathnayake opened the pull request:
…iews

ThashmikaRathnayake committed:
fix: updated device data fetching using relational joins instead of views

kavindalj commented on frontend/src/components/AllDevicesTable.jsx:
```diff
@@ -18,29 +18,61 @@ const AllDevicesTable = () => {
         const fetchdevicesData = async () => {
+
+            //To get the current colombo time 
+            const colomboTime = new Date().toLocaleString("en-US", { timeZone: "Asia/Colombo" });
```
Consider moving the hard-coded time zone ("Asia/Colombo") to an environment variable like REACT_APP_TIMEZONE for better configurability.

kavindalj reviewed (changes requested):
move hard-coded time zone to an environment variable for better configurability.

ThashmikaRathnayake committed:
refactor: fetch timezone from environment variable

ThashmikaRathnayake committed:
refactor: use show_id instead of id for clearer display in user_profiles

ThashmikaRathnayake committed:
feat: add formatted user ID display with prefix and zero-padding

Changed the user ID display from a plain numeric value (show_id) to a formatted string with a "USER" prefix and zero-padded numbers (Ex: USR001, USR002).

This improves readability and provides a clearer, more user-friendly identifier in the UI while keeping the original numeric ID intact for backend operations. The formatting is done in the frontend React code for flexibility without modifying the database schema.
